lib/facter/cisco_nexus.rb

```diff
@@ -0,0 +1,47 @@
+module Puppet::Facter
+  # class to handle the platform facts which are consistent between agent mode and agentless mode
+  class CiscoNexus
```

Comment:
the class `Puppet::Facter::CiscoNexus` should go to `lib/puppet/facter/cisco_nexus.rb`, or the file `lib/facter/cisco_nexus.rb` should contain the class `Facter::CiscoNexus`.

I'm leaning towards the shorter class name in this case.